File: app/src/main/java/com/example/kolomachinetest/data/Result.java
package com.example.kolomachinetest.data;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class Result implements Parcelable {

    @SerializedName("id")
    public int id;

    @SerializedName("name")
    public String name;

    @SerializedName("description")
    public String description;

    @SerializedName("modified")
    public String modified;

    @SerializedName("thumbnail")
    public Thumbnail thumbnail;

    @SerializedName("resourceURI")
    public String resourceURI;

    @SerializedName("comics")
    public Comics comics;

    @SerializedName("series")
    public Series series;

    @SerializedName("stories")
    public Stories stories;

    @SerializedName("events")
    public Events events;

    @SerializedName("urls")
    public List<Url> urls;

    @SerializedName("digitalId")
    public int digitalId;

    @SerializedName("title")
    public String title;

    @SerializedName("issueNumber")
    public int issueNumber;

    @SerializedName("variantDescription")
    public String variantDescription;

    @SerializedName("isbn")
    public String isbn;

    @SerializedName("upc")
    public String upc;

    @SerializedName("diamondCode")
    public String diamondCode;

    @SerializedName("ean")
    public String ean;

    @SerializedName("issn")
    public String issn;

    @SerializedName("format")
    public String format;

    @SerializedName("pageCount")
    public int pageCount;

    @SerializedName("textObjects")
    public List<TextObject> textObjects;

    @SerializedName("variants")
    public List<Series> variants;

    @SerializedName("collectedIssues")
    public List<Series> collectedIssues;

    @SerializedName("prices")
    public List<Price> prices;

    @SerializedName("images")
    public List<Thumbnail> images;

    @SerializedName("creators")
    public Creators creators;

    @SerializedName("characters")
    public Creators characters;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getModified() {
        return modified;
    }

    public void setModified(String modified) {
        this.modified = modified;
    }

    public Thumbnail getThumbnail() {
        return thumbnail;
    }

    public void setThumbnail(Thumbnail thumbnail) {
        this.thumbnail = thumbnail;
    }

    public String getResourceURI() {
        return resourceURI;
    }

    public void setResourceURI(String resourceURI) {
        this.resourceURI = resourceURI;
    }

    public Comics getComics() {
        return comics;
    }

    public void setComics(Comics comics) {
        this.comics = comics;
    }

    public Series getSeries() {
        return series;
    }

    public void setSeries(Series series) {
        this.series = series;
    }

    public Stories getStories() {
        return stories;
    }

    public void setStories(Stories stories) {
        this.stories = stories;
    }

    public Events getEvents() {
        return events;
    }

    public void setEvents(Events events) {
        this.events = events;
    }

    public List<Url> getUrls() {
        return urls;
    }

    public void setUrls(List<Url> urls) {
        this.urls = urls;
    }

    public int getDigitalId() {
        return digitalId;
    }

    public void setDigitalId(int digitalId) {
        this.digitalId = digitalId;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public int getIssueNumber() {
        return issueNumber;
    }

    public void setIssueNumber(int issueNumber) {
        this.issueNumber = issueNumber;
    }

    public String getVariantDescription() {
        return variantDescription;
    }

    public void setVariantDescription(String variantDescription) {
        this.variantDescription = variantDescription;
    }

    public String getIsbn() {
        return isbn;
    }

    public void setIsbn(String isbn) {
        this.isbn = isbn;
    }

    public String getUpc() {
        return upc;
    }

    public void setUpc(String upc) {
        this.upc = upc;
    }

    public String getDiamondCode() {
        return diamondCode;
    }

    public void setDiamondCode(String diamondCode) {
        this.diamondCode = diamondCode;
    }

    public String getEan() {
        return ean;
    }

    public void setEan(String ean) {
        this.ean = ean;
    }

    public String getIssn() {
        return issn;
    }

    public void setIssn(String issn) {
        this.issn = issn;
    }

    public String getFormat() {
        return format;
    }

    public void setFormat(String format) {
        this.format = format;
    }

    public int getPageCount() {
        return pageCount;
    }

    public void setPageCount(int pageCount) {
        this.pageCount = pageCount;
    }

    public List<TextObject> getTextObjects() {
        return textObjects;
    }

    public void setTextObjects(List<TextObject> textObjects) {
        this.textObjects = textObjects;
    }

    public List<Series> getVariants() {
        return variants;
    }

    public void setVariants(List<Series> variants) {
        this.variants = variants;
    }

    public List<Series> getCollectedIssues() {
        return collectedIssues;
    }

    public void setCollectedIssues(List<Series> collectedIssues) {
        this.collectedIssues = collectedIssues;
    }

    public List<Price> getPrices() {
        return prices;
    }

    public void setPrices(List<Price> prices) {
        this.prices = prices;
    }

    public List<Thumbnail> getImages() {
        return images;
    }

    public void setImages(List<Thumbnail> images) {
        this.images = images;
    }

    public Result() {
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(this.id);
        dest.writeString(this.name);
        dest.writeString(this.description);
        dest.writeString(this.modified);
        dest.writeParcelable(this.thumbnail, flags);
        dest.writeString(this.resourceURI);
        dest.writeParcelable(this.comics, flags);
        dest.writeParcelable(this.series, flags);
        dest.writeParcelable(this.stories, flags);
        dest.writeParcelable(this.events, flags);
        dest.writeTypedList(this.urls);
        dest.writeInt(this.digitalId);
        dest.writeString(this.title);
        dest.writeInt(this.issueNumber);
        dest.writeString(this.variantDescription);
        dest.writeString(this.isbn);
        dest.writeString(this.upc);
        dest.writeString(this.diamondCode);
        dest.writeString(this.ean);
        dest.writeString(this.issn);
        dest.writeString(this.format);
        dest.writeInt(this.pageCount);
        dest.writeList(this.textObjects);
        dest.writeTypedList(this.variants);
        dest.writeTypedList(this.collectedIssues);
        dest.writeTypedList(this.prices);
        dest.writeTypedList(this.images);
        dest.writeParcelable(this.creators, flags);
        dest.writeParcelable(this.characters, flags);
    }

    public void readFromParcel(Parcel source) {
        this.id = source.readInt();
        this.name = source.readString();
        this.description = source.readString();
        this.modified = source.readString();
        this.thumbnail = source.readParcelable(Thumbnail.class.getClassLoader());
        this.resourceURI = source.readString();
        this.comics = source.readParcelable(Comics.class.getClassLoader());
        this.series = source.readParcelable(Series.class.getClassLoader());
        this.stories = source.readParcelable(Stories.class.getClassLoader());
        this.events = source.readParcelable(Events.class.getClassLoader());
        this.urls = source.createTypedArrayList(Url.CREATOR);
        this.digitalId = source.readInt();
        this.title = source.readString();
        this.issueNumber = source.readInt();
        this.variantDescription = source.readString();
        this.isbn = source.readString();
        this.upc = source.readString();
        this.diamondCode = source.readString();
        this.ean = source.readString();
        this.issn = source.readString();
        this.format = source.readString();
        this.pageCount = source.readInt();
        this.textObjects = new ArrayList<>();
        source.readList(this.textObjects, TextObject.class.getClassLoader());
        this.variants = source.createTypedArrayList(Series.CREATOR);
        this.collectedIssues = source.createTypedArrayList(Series.CREATOR);
        this.prices = source.createTypedArrayList(Price.CREATOR);
        this.images = source.createTypedArrayList(Thumbnail.CREATOR);
        this.creators = source.readParcelable(Creators.class.getClassLoader());
        this.characters = source.readParcelable(Creators.class.getClassLoader());
    }

    protected Result(Parcel in) {
        this.id = in.readInt();
        this.name = in.readString();
        this.description = in.readString();
        this.modified = in.readString();
        this.thumbnail = in.readParcelable(Thumbnail.class.getClassLoader());
        this.resourceURI = in.readString();
        this.comics = in.readParcelable(Comics.class.getClassLoader());
        this.series = in.readParcelable(Series.class.getClassLoader());
        this.stories = in.readParcelable(Stories.class.getClassLoader());
        this.events = in.readParcelable(Events.class.getClassLoader());
        this.urls = in.createTypedArrayList(Url.CREATOR);
        this.digitalId = in.readInt();
        this.title = in.readString();
        this.issueNumber = in.readInt();
        this.variantDescription = in.readString();
        this.isbn = in.readString();
        this.upc = in.readString();
        this.diamondCode = in.readString();
        this.ean = in.readString();
        this.issn = in.readString();
        this.format = in.readString();
        this.pageCount = in.readInt();
        this.textObjects = new ArrayList<>();
        in.readList(this.textObjects, TextObject.class.getClassLoader());
        this.variants = in.createTypedArrayList(Series.CREATOR);
        this.collectedIssues = in.createTypedArrayList(Series.CREATOR);
        this.prices = in.createTypedArrayList(Price.CREATOR);
        this.images = in.createTypedArrayList(Thumbnail.CREATOR);
        this.creators = in.readParcelable(Creators.class.getClassLoader());
        this.characters = in.readParcelable(Creators.class.getClassLoader());
    }

    public static final Parcelable.Creator<Result> CREATOR = new Parcelable.Creator<Result>() {
        @Override
        public Result createFromParcel(Parcel source) {
            return new Result(source);
        }

        @Override
        public Result[] newArray(int size) {
            return new Result[size];
        }
    };
}

File: app/src/main/java/com/example/kolomachinetest/api/repo/marvel/data/Result.kt
package com.example.kolomachinetest.api.repo.marvel.data

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import com.example.kolomachinetest.api.repo.marvel.data.Thumbnail
import com.example.kolomachinetest.api.repo.marvel.data.Comics
import com.example.kolomachinetest.api.repo.marvel.data.Series
import com.example.kolomachinetest.api.repo.marvel.data.Stories
import com.example.kolomachinetest.api.repo.marvel.data.TextObject
import com.example.kolomachinetest.api.repo.marvel.data.Price
import com.example.kolomachinetest.api.repo.marvel.data.Creators
import android.os.Parcel
import java.util.ArrayList

class Result : Parcelable {
    @SerializedName("id")
    var id = 0

    @SerializedName("name")
    var name: String? = null

    @SerializedName("description")
    var description: String? = null

    @SerializedName("modified")
    var modified: String? = null

    @SerializedName("thumbnail")
    var thumbnail: Thumbnail? = null

    @SerializedName("resourceURI")
    var resourceURI: String? = null

    @SerializedName("comics")
    var comics: Comics? = null

    @SerializedName("series")
    var series: Series? = null

    @SerializedName("stories")
    var stories: Stories? = null

    @SerializedName("events")
    var events: Events? = null

    @SerializedName("urls")
    var urls: List<Url>? = null

    @SerializedName("digitalId")
    var digitalId = 0

    @SerializedName("title")
    var title: String? = null

    @SerializedName("issueNumber")
    var issueNumber = 0

    @SerializedName("variantDescription")
    var variantDescription: String? = null

    @SerializedName("isbn")
    var isbn: String? = null

    @SerializedName("upc")
    var upc: String? = null

    @SerializedName("diamondCode")
    var diamondCode: String? = null

    @SerializedName("ean")
    var ean: String? = null

    @SerializedName("issn")
    var issn: String? = null

    @SerializedName("format")
    var format: String? = null

    @SerializedName("pageCount")
    var pageCount = 0

    @SerializedName("textObjects")
    var textObjects: List<TextObject?>? = null

    @SerializedName("variants")
    var variants: List<Series>? = null

    @SerializedName("collectedIssues")
    var collectedIssues: List<Series>? = null

    @SerializedName("prices")
    var prices: List<Price>? = null

    @SerializedName("images")
    var images: List<Thumbnail>? = null

    @SerializedName("creators")
    var creators: Creators? = null

    @SerializedName("characters")
    var characters: Creators? = null

    constructor() {}

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(id)
        dest.writeString(name)
        dest.writeString(description)
        dest.writeString(modified)
        dest.writeParcelable(thumbnail, flags)
        dest.writeString(resourceURI)
        dest.writeParcelable(comics, flags)
        dest.writeParcelable(series, flags)
        dest.writeParcelable(stories, flags)
        dest.writeParcelable(events, flags)
        dest.writeTypedList(urls)
        dest.writeInt(digitalId)
        dest.writeString(title)
        dest.writeInt(issueNumber)
        dest.writeString(variantDescription)
        dest.writeString(isbn)
        dest.writeString(upc)
        dest.writeString(diamondCode)
        dest.writeString(ean)
        dest.writeString(issn)
        dest.writeString(format)
        dest.writeInt(pageCount)
        dest.writeList(textObjects)
        dest.writeTypedList(variants)
        dest.writeTypedList(collectedIssues)
        dest.writeTypedList(prices)
        dest.writeTypedList(images)
        dest.writeParcelable(creators, flags)
        dest.writeParcelable(characters, flags)
    }

    fun readFromParcel(source: Parcel) {
        id = source.readInt()
        name = source.readString()
        description = source.readString()
        modified = source.readString()
        thumbnail = source.readParcelable(Thumbnail::class.java.classLoader)
        resourceURI = source.readString()
        comics = source.readParcelable(Comics::class.java.classLoader)
        series = source.readParcelable(Series::class.java.classLoader)
        stories = source.readParcelable(Stories::class.java.classLoader)
        events = source.readParcelable(Events::class.java.classLoader)
        urls = source.createTypedArrayList(Url.CREATOR)
        digitalId = source.readInt()
        title = source.readString()
        issueNumber = source.readInt()
        variantDescription = source.readString()
        isbn = source.readString()
        upc = source.readString()
        diamondCode = source.readString()
        ean = source.readString()
        issn = source.readString()
        format = source.readString()
        pageCount = source.readInt()
        textObjects = ArrayList()
        source.readList(textObjects as ArrayList<TextObject?>, TextObject::class.java.classLoader)
        variants = source.createTypedArrayList(Series.CREATOR)
        collectedIssues = source.createTypedArrayList(Series.CREATOR)
        prices = source.createTypedArrayList(Price.CREATOR)
        images = source.createTypedArrayList(Thumbnail.CREATOR)
        creators = source.readParcelable(Creators::class.java.classLoader)
        characters = source.readParcelable(Creators::class.java.classLoader)
    }

    protected constructor(`in`: Parcel) {
        id = `in`.readInt()
        name = `in`.readString()
        description = `in`.readString()
        modified = `in`.readString()
        thumbnail = `in`.readParcelable(Thumbnail::class.java.classLoader)
        resourceURI = `in`.readString()
        comics = `in`.readParcelable(Comics::class.java.classLoader)
        series = `in`.readParcelable(Series::class.java.classLoader)
        stories = `in`.readParcelable(Stories::class.java.classLoader)
        events = `in`.readParcelable(Events::class.java.classLoader)
        urls = `in`.createTypedArrayList(Url.CREATOR)
        digitalId = `in`.readInt()
        title = `in`.readString()
        issueNumber = `in`.readInt()
        variantDescription = `in`.readString()
        isbn = `in`.readString()
        upc = `in`.readString()
        diamondCode = `in`.readString()
        ean = `in`.readString()
        issn = `in`.readString()
        format = `in`.readString()
        pageCount = `in`.readInt()
        textObjects = ArrayList()
        `in`.readList(textObjects as ArrayList<TextObject?>, TextObject::class.java.classLoader)
        variants = `in`.createTypedArrayList(Series.CREATOR)
        collectedIssues = `in`.createTypedArrayList(Series.CREATOR)
        prices = `in`.createTypedArrayList(Price.CREATOR)
        images = `in`.createTypedArrayList(Thumbnail.CREATOR)
        creators = `in`.readParcelable(Creators::class.java.classLoader)
        characters = `in`.readParcelable(Creators::class.java.classLoader)
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<Result> = object : Parcelable.Creator<Result> {
            override fun createFromParcel(source: Parcel): Result {
                return Result(source)
            }

            override fun newArray(size: Int): Array<Result?> {
                return arrayOfNulls(size)
            }
        }
    }
}